### DISCUSSION TITLE
GoPro keeps turning on when connected to external power source.

### DISCUSSION THREAD

**Discussion by LeoLopesLage** [gopro/labs#150] [Q&A]
I'm running the construction scrip with the U! command to sync with my cloud subscription.
I found that it only upload when connected to an external power source. So I did it. Worked and everything is in sync.
But, the GoPro keeps turning on every 5 seconds because it is connected to that external source... Than no buttons is pressed and the script put it to sleep. And it turns on again... loop.
Does anyone know how to fix it?
Thanks

**Comment by dnewman-gpsw** [comment]
Share the script and all the metadata from the last MP4 captured using https://gopro.github.io/labs/control/metadata/  Also good to say what camera model (although the metadata will show that.)

This is not the default behavior, so might be your configuration.

**Reply by LeoLopesLage** [reply]
Hi. 
Below is two metadata from the last two files (a timelapse video and a picture) from this GoPro.
It is a Hero 10 Black.
The  script used is this one: mPdP>07:59<10:40!81SQ~!08:00S!U!1R

Thanks for your help.

1) Timelapse video metadata:

DEVC 
  DVID 1
  DVNM Global Settings
  VERS 8, 2, 0
  FMWR H21.01.01.46.70
  LINF LKO1090802401251
  CINF 189, 63, 205, 28, 122, 207, 151, 166, 210, 33, 129, 254, 165, 103, 133, 36
  CASN C3464250131765
  MINF HERO10 Black
  MUID 483213245, 2794966906, 4269875666, 612722597, 0, 0, 0, 0
  CMOD 13
  MTYP 1
  OREN U
  DZOM Y
  DZST 0
  SMTR N
  PRTN Y
  PTWB AUTO
  PTSH MED
  PTCL NATURAL
  EXPT AUTO
  PIMX 1600
  PIMN 100
  PTEV 0.0
  RATE 1_10SEC
  VFOV L
  SROT 18.3240
  ZFOV 100.5979
  EISE N
  EISA N/A
  HLVL N
  AUPT N
  AUDO .empty.
  BROD .empty.
  BRID 0
  PVUL F
  PRJT GPRO
  SOFF 0
  CLKS 2
  CDAT 0x00000000636D270B
  SCTM 0
  PRNA 8
  PRNU 0
  SCAP N
  CDTM 0
  DUST NO_LIMIT
  VRES 3840, 2160
  VFPS 30000, 1001
  HSGT OFF
  BITR STANDARD
  MMOD STEREO
  RAMP .empty.
  TZON 32767
  CLKC 0
  DZMX 1.4000
  ORDP Y
  CLDP Y
DEVC 
  DVID USRM
  DVNM User Metadata
  USRD 1
DEVC 
  DVID HLMT
  DVNM Highlights

2) Picture metadata:

DEVC 
  DVID 1
  DVNM Global Settings
  VERS 8, 2, 0
  FMWR H21.01.01.46.70
  LINF LKO1090802401251
  CINF 189, 63, 205, 28, 122, 207, 151, 166, 210, 33, 129, 254, 165, 103, 133, 36
  CASN C3464250131765
  MINF HERO10 Black
  MUID 483213245, 2794966906, 4269875666, 612722597, 0, 0, 0, 0
  CMOD 16
  MTYP 13
  OREN U
  DZOM Y
  DZST 0
  ZFOV 0.0000
  SMTR N
  PRTN Y
  PTWB AUTO
  PTSH MED
  PTCL GOPRO
  EXPT AUTO
  PIMX 3200
  PIMN 100
  PTEV 0.0
  RATE 1
  HLVL N
  PRES 23MP_L
  PHDR S_STILL
  PRAW N
  HFLG 0
  PVUL F
  SOFF 0
  PRJT GPRO
  CLKS 1
  GRAV 0.0025, 0.9999, 0.0118
  IORI 1.0000, 0.0000, 0.0000, 0.0000
  CORI 1.0000, 0.0000, 0.0000, 0.0000
  CDAT 0x00000000636E24C8
  SCTM 0
  PRNA 3
  PRNU 0
  SCAP N
  CDTM 3000
  DUST .empty.
  TZON 32767
  CLKC 0
  DZMX 2.0000
  PTMP 3087
  ORDP Y
DEVC 
  DVID USRM
  DVNM User Metadata
  USRD 1
  ÿÛ 
    

**Comment by dnewman-gpsw** [comment]
The metadata looks fine. I expect your power source might not be delivering sufficient current (2A required) or it is auto sensing (most Lithium Ion power bricks do this) and turning on and off based on power load.  In MISC/qrlog.txt you can see what the camera is doing.  I ran your script on a HERO10 and it works fine (with external power - using an always on Voltaic V50.)  This is a sample from my log file:


[2022-11-21 09:20:04] Woke with Alarm:[09:20m01]!SQ~!08:00S!U!1R 
[2022-11-21 09:20:07] Labs:Main Start 
[2022-11-21 09:20:07] Labs:Initialize 
[2022-11-21 09:20:07] Woke with Command:[09:20m01]!SQ~!08:00S!U!1R 
[2022-11-21 09:20:07] Start Now:Past by 7 seconds 
[2022-11-21 09:20:07] Timer wake command:!SQ~!08:00S!U!1R 
[2022-11-21 09:20:09] skip_command:!U!1R 
[2022-11-21 09:20:10] else skipped, process from:!U!1R 
[2022-11-21 09:20:10] upload:done 
[2022-11-21 09:20:14] true 09:20 > 07:59:>07:59<10:40!81SQ~!08:00S!U!1R 
[2022-11-21 09:20:14] true 09:20 < 10:40:<10:40!81SQ~!08:00S!U!1R 
[2022-11-21 09:20:17] Shutdown and Sleep:76 seconds 
[2022-11-21 09:20:17] shutdown:rtc wake scheduled 
[2022-11-21 09:21:38] Woke with Alarm:[09:21m01]!SQ~!08:00S!U!1R 
[2022-11-21 09:21:41] Labs:Main Start 
[2022-11-21 09:21:41] Labs:Initialize 
...

However, I think there is a bug in your script, as you should upload with every phone (slow), just upload once a day.

mPdP>07:59<10:40!81SQ+!1R~!08:00N!U!1R

This will upload at 8am each day.

**Reply by LeoLopesLage** [reply]
Thank you. Will try your script and change the external source in the next test.
As you mention, I am having trouble uploading to the cloud. Can't make it work either.
When I turn it on and go to settings manually, the connections are always off. So I turn it on and sync manually. Maybe it will work with yours.
Will let you know. 
Thank you!

**Reply by dnewman-gpsw** [reply]
Practice with https://gopro.github.io/labs/control/dailytl/ as that worked the last time I tried it.

**Reply by LeoLopesLage** [reply]
Thank you.
You nailled the external power source problem. Tryed with a AC power and worked fine.
Trying to solve the upload problem. Still doesn't upload the files.

**Reply by dnewman-gpsw** [reply]
Try the camera menu for a manual upload (Swipe down for menus, swipe to right menu, preferences -> auto upload -> Manual Upload.) If that doesn't work, Factory Reset your camera and re-pair with the Quik App and setup auto update again. Labs needs a working auto upload to be running. !U is simply the QR command that start a manual upload.

**Reply by LeoLopesLage** [reply]
Manual upload works fine!
Labs upload starts and the sleep counter starts too. It goes to sleep in 3 to 5s and doesn't upload a single file.
What can it be now?

**Reply by dnewman-gpsw** [reply]
Using only !U command, does it work?

**Reply by LeoLopesLage** [reply]
Hi. I did a small video showing what's happening. Hope you can help me. 
https://user-images.githubusercontent.com/91095702/203427718-a9156651-511a-4c43-88c9-e7c8eb14f2a9.MOV

**Reply by dnewman-gpsw** [reply]
You are still running a script, so you need to stop that before testing. That is why is says "button to cancel", press a button. Then run !U alone.

**Reply by LeoLopesLage** [reply]
Ok. Cleared all scripts. Did a preset reset. Scanned the !U alone QR code. Worked!
![image](https://user-images.githubusercontent.com/91095702/203549253-22a3fec3-e7e8-4bf1-b5ae-90e06201f034.jpeg)

Now, why it doesn't work with the script you've sent me?

**Reply by LeoLopesLage** [reply]
Trying a new one. Will post the results soon. Thank you

**Reply by LeoLopesLage** [reply]
!U only works alone.
If used in any other script it simply tries to connect and the camera shuts down before it does.
Will try with !L command to load a script just to upload. But need to learn how it works before. Where do I put yhe script?

**Reply by dnewman-gpsw** [reply]
Learn from scripts that are know to work https://gopro.github.io/labs/control/dailytl/  Look at the log file and learn from that.

**Reply by LeoLopesLage** [reply]
Where is the log file saved?
Can't find anything about the!Lscript command. What's the file type and where do I put it so I can load with that command?
Thank you for you help and patience.

**Reply by dnewman-gpsw** [reply]
Log file: MISC/qrlog.txt
Scripts: https://gopro.github.io/labs/control/actions/  But you are not ready for saving and loading sd card based scripts, you only use those after you have things working and you want more complexity.

**Reply by LeoLopesLage** [reply]
So, "learned" the code this past month.
Tried a lot of working scripts and custom ones.
The !U command seems to not work with recursion... if you make a script to take one photo and upload it works. If you ask to take multiples photos in an interval an upload each one... if doesn't. 

Tried with "if else" commands, it doesn't respond well. Some times works, and the day after, it doesn't start the GoPro.

Tried a lot of combination... none worked.

So, I'm giving up the !U command. Will make the script only to take the pictures in that interval and once a week go to the sight and upload manually.

Thank you for you help.

**Comment by jo196605** [comment]
Are you using the original gopro usb cable? Some cables give not enough power and than the gopro keeps thurning on...

**Reply by LeoLopesLage** [reply]
Thanks. This one was solved. 
The problem now is the !U command.
